[SMP-1345] SMP 0.8.0 release notes (#2585)
harness/developer-hub: release-notes/self-managed-enterprise-edition.md

  
  To execute a pipeline with Input Set references, there is an optional field, `lastYamlToMerge` in the API request body. The values set for the `lastYamlToMerge` field were not acknowledged.

  This issue is fixed by acknowledging and properly overriding the `lastYamlToMerge` value. For more details, go to [Execute a pipeline with input set references](https://apidocs.harness.io/tag/Pipeline-Execute#operation/postPipelineExecuteWithInputSetList).

- Jenkins step marking voluntary settings as mandatory. (CDS-70071, ZD-44924)
  
  User were unable to save empty values for job parameters in the Jenkins step due to validations present in the UI. This has been fixed now and the incorrect validations have been removed.

- Fixed an issue to honor default values in runtime inputs when trying to reconcile. (CDS-70075, ZD-44892)

- Unable to trigger pipelines using custom webhook triggers because the latest enhancement to custom webhook triggers requires a unique token for authentication. (CDS-70320, ZD-45022)
  

>   This issue was caused by introducing a new API for custom webhook triggers, `/v3`, that generates a token for authenticating webhook triggers. You can see the token when you copy the trigger cURL command in Harness. This issues is fixed by allowing users to continue to use the previous API, `v2`, when the feature flag `SPG_DISABLE_CUSTOM_WEBHOOK_V3_URL` is enabled.

Reviewer: ```suggestion
  This issue was caused by introducing a new API for custom webhook triggers, `/v3`, that generates a token for authenticating webhook triggers. You can see the token when you copy the trigger cURL command in Harness. This issue is fixed by allowing users to continue to use the previous API, `v2`, when the feature flag `SPG_DISABLE_CUSTOM_WEBHOOK_V3_URL` is enabled.
```

- Previously, there was an edge case where an extremely large number of pages in a UI listing could cause performance degradation. This issue has been fixed, and page numbering is now disabled if the page count exceeds 1000 pages. (FFM-7993)

- Previously, a user could select Feature Flags in Harness without having a license, and would then get a 404 error. Now, users only see the FF module if they have an active license (including a free license). (FFM-8002)

- The **Create a new flag** button did not behave as expected on the onboarding screen if there were existing flags. This issue has been fixed. (FFM-8019)

#### Harness Delegate

- The account-level **Session Timeout (in minutes)** allowed values greater than the 4320 minute maximum. (PL-32498)

   This issue has been resolved by adding a code validation. The field no longer accepts values above 4320 minutes.

   This item requires Harness Delegate version 79707. For information about features that require a specific delegate version, go to the [Delegate release notes](/release-notes/delegate).

> - Enhanced handling and logging for the `No enum constant io.harness.delegate.message.MessengerType.WATCHEIN` exception to enable the actual malformed message. This error indicates that a message is malformed and only occurs when there is an error during writing, for example, out of disk, process killed, etc. (PL-38245)

Reviewer: ```suggestion
- Enhanced handling and logging for the `No enum constant io.harness.delegate.message.MessengerType.WATCHEIN` exception to enable the actual malformed message. This error indicates that a message is malformed and only occurs when there is a write error such as out of disk, process killed, etc. (PL-38245)
```

Review the notes below for details about recent changes to Harness Self-Managed Enterprise Edition, NextGen. For release notes for FirstGen Self-Managed Enterprise Edition, go to [Self-Managed Enterprise Edition release notes (FirstGen)](/docs/first-gen/firstgen-release-notes/harness-on-prem-release-notes). Additionally, Harness publishes security advisories for every release. Go to the [Harness Trust Center](https://trust.harness.io/?itemUid=c41ff7d5-98e7-4d79-9594-fd8ef93a2838&source=documents_card) to request access to the security advisories.

## Latest - July 31, 2023, version 79819

### Known issues

-  Terraform-based pipeline automation does not function correctly in Istio environments. UI operations are not affected. (PL-39776)

> - The `log-service` created separate Redis streams and set the expiration of all keys. Harness has temporarily disabled Redis in the `log-service` for High availability configuration. (CI-9000)

Reviewer 2: I would say - The `log-service` created separate Redis streams and set the expiration of all keys. Harness has temporarily does not support High availability configuration for the `log-service` till this issue is resolved

Author: @akshay-khole @pratmit added a commit to address. Thank you both for your review.

```mdx-code-block
<Tabs>
  <TabItem value="What's new">
```

This release includes the following Harness module and component versions.

| **Name** | **Version** |
| :-- | :--: |
| Helm Chart | [0.8.0](https://github.com/harness/helm-charts/releases/tag/harness-0.8.0) |
| Air Gap Bundle | [0.8.0](https://console.cloud.google.com/storage/browser/smp-airgap-bundles/harness-0.8.0) |
| NG Manager | 79819 |
| CI Manager | 4902 |
| Pipeline Service | 1.37.12 |
| Platform Service | 79601 |
| Access Control Service | 79400 |
| Change Data Capture | 79819 |
| Test Intelligence Service | release-177 |
| NG UI | 0.353.10 |
| LE NG | 68004 |

#### Self-Managed Enterprise Edition

- Harness now supports external self-managed databases for high availability. (SMP-1577, SMP-1617, SMP-1645, SMP-1646)

  To learn how to configure an external self-managed database, go to the following tutorials.

<!---

>   - [MongoDB](/tutorials/self-managed-enterprise-edition/use-an-external-self-managed-mongodb/)

Reviewer 2: will these URLs be replaced ?

-  Terraform-based pipeline automation does not function correctly in Istio environments. UI operations are not affected. (PL-39776)

> - The `log-service` created separate Redis streams and set the expiration of all keys. Harness temporarily does note support high availability configuration for the `log-service` until this issue is resolved. (CI-9000)

Reviewer 2: note -> not

Author: thanks, @akshay-khole
